lib/active_record_inherit_assoc.rb

```diff
@@ -25,10 +25,29 @@ def attribute_inheritance_hash
   end
 end
 
-ActiveRecord::Associations::Association.class_eval do
-  prepend ActiveRecordInheritAssocPrepend
+ActiveRecord::Associations::Association.send(:prepend, ActiveRecordInheritAssocPrepend)
+
+module ActiveRecordInheritPreloadAssocPrepend
+  def associated_records_by_owner(*args)
+    super.tap do |result|
+      next unless reflection.options[:inherit]
```

Comment:
having both methods rely on reflection.options[:inherit] is a bit entangling, how about:

```Ruby
next unless inherit = reflection.options[:inherit]
result.each do |owner, associated_records|
  filter_associated_records_with_inherit!(inherit, owner, associated_records)
end
```